darwin/dataset/download_manager.py

```diff
@@ -339,7 +344,7 @@ def _download_single_slot_from_json_annotation(
             image_url = image["url"]
             image_filename = image["file_name"]
             suffix = Path(image_filename).suffix
-            stem = annotation_path.stem
+            stem = Path(annotation.filename).stem
```

Comment:
In cases where there are duplicate item names, the `annotation_path.stem` will have an `_n` appended, which will cause darwin-py to fail to find the corresponding item. Therefore, use the item name since this will not have a `_n` appended for uniqueness